pandas/core/arrays/integer.py

```diff
@@ -95,6 +95,15 @@ def construct_array_type(cls) -> Type["IntegerArray"]:
         """
         return IntegerArray
 
+    def _get_common_type(self, dtypes: List[DtypeObj]) -> Optional[DtypeObj]:
+        # for now only handle other integer types
+        if not all(isinstance(t, _IntegerDtype) for t in dtypes):
+            return None
+        np_dtype = np.find_common_type([t.numpy_dtype for t in dtypes], [])
+        if np.issubdtype(np_dtype, np.integer):
+            return _dtypes[str(np_dtype)]
+        return None
```

Comment:
For now, this only handles concatting of several EA IntegerDtypes. Of course, we can expand this to also handle mix of EA Int dtype with numpy int dtypes, and also handle mix of integer dtype with non-integer (like coercing bool to integer, coercing to floats, ...). 
But this could also be done separately in a follow-up  to fully flesh this out, once the protocol is in place (and at least the above implements the minimal to show it works).

Reply:
Agreed with keeping it simple for now. We can try to achieve the behavior on master (aside from fixing a few bugs) and then discuss whether we should handle casting.

Reply:
agreed

Reply:
sgtm